ls1intum/Artemis
# [Feature] Overwrite timeout message feedback

### Checklist

### Motivation and Context


In case of timeout exceptions (e.g. due to endless while(true) loops in student code), the error message just states, that there was a timeout after x seconds. We want to show a more verbose text, so that the student knows what to look for.
Fixes: #2328 


### Description

In case a time out warning occurs, we are replacing the text with a more descriptive one. We check with regex if the warning contains some sort of a `.*TimeoutException`. 

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Create any programming exercise
4. Participate in the exercise and cause for example an endless while loop (while-true)
5. The automatic testcase feedback should mention that a timeout has occurred.

### Test Coverage



No changes.

### Screenshots


#### Feedback with timeout message:
![image](https://user-images.githubusercontent.com/33484318/110248621-57256580-7f72-11eb-9c7f-7d1ce8ff12a6.png)
